articles/integrating-graphql-api-in-a-django-application/index.md

```diff
@@ -0,0 +1,492 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/integrating-graphql-api-in-a-django-application/
+title: Integrating GraphQL API in a Django application
+description: This article provides a guide to using GraphQL in a Django application.
+author: anita-achu
+date: 2021-05-16T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images: 
+
+  - url: /engineering-education/integrating-graphql-api-in-a-django-application/hero.jpg
+    alt: graphql+django
+---
+
+
+GraphQL is an open-source query language used to communicate data between the client (the browser) and the server. As explained in [GraphQL doc](https://graphql.org/)s, "GraphQL is a query language for APIs and a runtime for fulfilling those queries with your existing data..."
+
+Created by Facebook in 2012, GraphQL provides a runtime environment for Application Program Interfaces (API) which is easy to use, fast and developer-friendly. Over time, GraphQL has gained a wide range of use by companies such as Microsoft, Github, Shopify, Amazon, etc.
+
+The most outstanding attribute of GraphQL is that it returns the requested data from multiple sources with just a single request this makes it more preferred than RESTAPI which is the alternative.
+
+In essence, the client can request only the data they need in the response, which means you have complete control over the data structure, and in a single request, you can access a large number of services. GraphQL has a lot of advantages however, the features of GraphQL are outside of the scope of this tutorial. 
+
+In this tutorial, our focus would be on integrating a GraphQL API into a Django project and effectively using it to query data.
+
+### Prerequisites:
+
+- Basic knowledge of python
+- Basic knowledge of Django

```

Comment:
Change this to you need a good understanding of Django before proceeding with this article since you are not explaining the Django code snippets a lot.